File: count_pages.py
import re
import requests
from urllib.parse import urljoin, urlparse
from collections import defaultdict
import time

# Base URL
base_url = "https://unblocked-games.s3.amazonaws.com/"

# Read the dublix.html file
with open('dublix.html', 'r', encoding='utf-8') as file:
    html_content = file.read()

# Extract all links from the HTML
link_pattern = r'href=["\']([^"\']+)["\']'
links = re.findall(link_pattern, html_content)

# Filter and normalize links
internal_links = set()
external_links = set()

for link in links:
    # Skip javascript, mailto, and other non-http links
    if link.startswith('javascript:') or link.startswith('mailto:') or link.startswith('#'):
        continue
    
    # Convert relative links to absolute
    if link.startswith('./') or link.startswith('/'):
        absolute_link = urljoin(base_url, link)
    else:
        absolute_link = link
    
    # Check if it's an internal link (same domain)
    parsed_base = urlparse(base_url)
    parsed_link = urlparse(absolute_link)
    
    if parsed_base.netloc == parsed_link.netloc:
        # Clean up the URL - remove fragments
        clean_link = parsed_link._replace(fragment="").geturl()
        internal_links.add(clean_link)
    else:
        external_links.add(absolute_link)

# Categorize internal links
page_types = defaultdict(list)
other_files = set()

for link in internal_links:
    parsed = urlparse(link)
    path = parsed.path
    
    # Categorize by file type or directory
    if path.endswith('.html'):
        if path.endswith('index.html'):
            if path == '/' or path.endswith('/index.html'):
                page_types['Home Pages'].append(link)
            else:
                page_types['Category Index Pages'].append(link)
        else:
            page_types['Game Pages'].append(link)
    elif path.endswith('.css'):
        other_files.add('CSS Files')
    elif path.endswith('.js'):
        other_files.add('JavaScript Files')
    elif path.endswith('.webp') or path.endswith('.jpg') or path.endswith('.png') or path.endswith('.ico') or path.endswith('.svg'):
        other_files.add('Image Files')
    elif path.endswith('.json'):
        other_files.add('JSON Files')
    else:
        page_types['Other Pages'].append(link)

# Create a markdown report
report = f"""# Website Page Count Analysis

## Summary
- Total Internal Links: {len(internal_links)}
- Total External Links: {len(external_links)}
- Total Unique Pages: {sum(len(pages) for pages in page_types.values())}

## Page Categories

### Home Pages ({len(page_types['Home Pages'])})
{chr(10).join(f"- {link}" for link in sorted(page_types['Home Pages']))}

### Game Pages ({len(page_types['Game Pages'])})
{chr(10).join(f"- {link}" for link in sorted(page_types['Game Pages']))}

### Category Index Pages ({len(page_types['Category Index Pages'])})
{chr(10).join(f"- {link}" for link in sorted(page_types['Category Index Pages']))}

### Other Pages ({len(page_types['Other Pages'])})
{chr(10).join(f"- {link}" for link in sorted(page_types['Other Pages']))}

## Other File Types
{chr(10).join(f"- {file_type}" for file_type in sorted(other_files))}

## External Links ({len(external_links)})
{chr(10).join(f"- {link}" for link in sorted(external_links)[:20])}
{'...' if len(external_links) > 20 else ''}

## Total Web Pages
The website contains approximately **{sum(len(pages) for pages in page_types.values())} web pages**.
"""

# Save the report to a markdown file
with open('website_page_count.md', 'w', encoding='utf-8') as file:
    file.write(report)

print("Analysis complete. Report saved to website_page_count.md")
print(f"Total web pages found: {sum(len(pages) for pages in page_types.values())}")

File: count_pages.js
const fs = require('fs');
const path = require('path');

// 读取dublix.html文件
const htmlContent = fs.readFileSync('dublix.html', 'utf8');

// 提取所有链接的正则表达式
const linkPattern = /href=["']([^"']+)["']/g;
const links = [];
let match;

while ((match = linkPattern.exec(htmlContent)) !== null) {
    links.push(match[1]);
}

// 过滤和规范化链接
const internalLinks = new Set();
const externalLinks = new Set();
const baseUrl = "https://unblocked-games.s3.amazonaws.com/";

links.forEach(link => {
    // 跳过javascript、mailto和其他非http链接
    if (link.startsWith('javascript:') || link.startsWith('mailto:') || link.startsWith('#')) {
        return;
    }
    
    // 将相对链接转换为绝对链接
    let absoluteLink;
    if (link.startsWith('./') || link.startsWith('/')) {
        absoluteLink = new URL(link, baseUrl).href;
    } else {
        absoluteLink = link;
    }
    
    // 检查是否是内部链接（相同域名）
    try {
        const baseUrlObj = new URL(baseUrl);
        const linkObj = new URL(absoluteLink);
        
        if (baseUrlObj.hostname === linkObj.hostname) {
            // 清理URL - 移除片段
            linkObj.hash = '';
            internalLinks.add(linkObj.href);
        } else {
            externalLinks.add(absoluteLink);
        }
    } catch (e) {
        // 如果URL解析失败，跳过
        console.log('无法解析URL:', absoluteLink);
    }
});

// 分类内部链接
const pageTypes = {
    'Home Pages': [],
    'Game Pages': [],
    'Category Index Pages': [],
    'Other Pages': []
};
const otherFiles = new Set();

internalLinks.forEach(link => {
    try {
        const linkObj = new URL(link);
        const path = linkObj.pathname;
        
        // 按文件类型或目录分类
        if (path.endsWith('.html')) {
            if (path.endsWith('index.html')) {
                if (path === '/' || path.endsWith('/index.html')) {
                    pageTypes['Home Pages'].push(link);
                } else {
                    pageTypes['Category Index Pages'].push(link);
                }
            } else {
                pageTypes['Game Pages'].push(link);
            }
        } else if (path.endsWith('.css')) {
            otherFiles.add('CSS Files');
        } else if (path.endsWith('.js')) {
            otherFiles.add('JavaScript Files');
        } else if (path.match(/\.(webp|jpg|png|ico|svg)$/)) {
            otherFiles.add('Image Files');
        } else if (path.endsWith('.json')) {
            otherFiles.add('JSON Files');
        } else {
            pageTypes['Other Pages'].push(link);
        }
    } catch (e) {
        console.log('无法处理链接:', link);
    }
});

// 创建markdown报告
const totalPages = Object.values(pageTypes).reduce((sum, pages) => sum + pages.length, 0);
const report = `# 网站页面数量分析

## 摘要
- 内部链接总数: ${internalLinks.size}
- 外部链接总数: ${externalLinks.size}
- 唯一页面总数: ${totalPages}

## 页面分类

### 首页 (${pageTypes['Home Pages'].length})
${pageTypes['Home Pages'].sort().map(link => `- ${link}`).join('\n')}

### 游戏页面 (${pageTypes['Game Pages'].length})
${pageTypes['Game Pages'].sort().map(link => `- ${link}`).join('\n')}

### 分类索引页面 (${pageTypes['Category Index Pages'].length})
${pageTypes['Category Index Pages'].sort().map(link => `- ${link}`).join('\n')}

### 其他页面 (${pageTypes['Other Pages'].length})
${pageTypes['Other Pages'].sort().map(link => `- ${link}`).join('\n')}

## 其他文件类型
${Array.from(otherFiles).sort().map(fileType => `- ${fileType}`).join('\n')}

## 外部链接 (${externalLinks.size})
${Array.from(externalLinks).sort().slice(0, 20).map(link => `- ${link}`).join('\n')}
${externalLinks.size > 20 ? '...' : ''}

## 网页总数
该网站大约包含 **${totalPages} 个网页**。
`;

// 将报告保存到markdown文件
fs.writeFileSync('website_page_count.md', report, 'utf8');

console.log('分析完成。报告已保存到 website_page_count.md');
console.log(`找到的网页总数: ${totalPages}`);